feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/component/FilterChipWithDropdownMenu.kt

```diff
@@ -0,0 +1,145 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+
+data class SearchFilterItemUiState<T>(
+    val selectedItems: List<T>,
+    val items: List<T>,
+    val isSelected: Boolean = false,
+    val selectedValues: String = "",
+)
+
+const val FilterChipWithDropdownMenuItemTestTag = "FilterChipWithDropdownMenuItem"
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> FilterChipWithDropdownMenu(
+    searchFilterItemUiState: SearchFilterItemUiState<T>,
+    onSelected: (T, Boolean) -> Unit,
+    filterChipLabel: @Composable () -> Unit,
+    dropdownMenuItemText: @Composable (T) -> Unit,
+    modifier: Modifier = Modifier,
+    filterChipLeadingIcon: @Composable (() -> Unit)? = null,
+    filterChipTrailingIcon: @Composable (() -> Unit)? = null,
+    onFilterChipClick: (() -> Unit)? = null,
+    dropdownMenuItemLeadingIcon: @Composable ((T) -> Unit)? = null,
+    dropdownMenuItemTrailingIcon: @Composable ((T) -> Unit)? = null,
+) {
+    var expanded by remember { mutableStateOf(false) }
+    val onSelectedUpdated by rememberUpdatedState(newValue = onSelected)
+
+    val expandMenu = { expanded = true }
+    val shrinkMenu = { expanded = false }
+
+    val selectedItems = searchFilterItemUiState.selectedItems
+
+    Box(
+        modifier = modifier.wrapContentSize(Alignment.TopStart),
+    ) {
+        FilterChip(
+            selected = searchFilterItemUiState.isSelected,
+            onClick = {
+                onFilterChipClick?.invoke()
+                expandMenu()
+            },
+            label = filterChipLabel,
+            leadingIcon = filterChipLeadingIcon,
+            trailingIcon = filterChipTrailingIcon,
+        )
+        DropdownMenu(
+            expanded = expanded,
+            onDismissRequest = shrinkMenu,
+        ) {
+            searchFilterItemUiState.items.forEach { item ->
+                DropdownMenuItem(
+                    text = {
+                        dropdownMenuItemText(item)
+                    },
+                    leadingIcon = dropdownMenuItemLeadingIcon?.let { icon ->
+                        { icon(item) }
+                    },
+                    trailingIcon = dropdownMenuItemTrailingIcon?.let { icon ->
+                        { icon(item) }
+                    },
+                    onClick = {
+                        onSelectedUpdated(
+                            item,
+                            selectedItems.contains(item).not(),
+                        )
+                        shrinkMenu()
+                    },
+                    modifier = Modifier.testTag(FilterChipWithDropdownMenuItemTestTag),
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun <T> FilterChipWithDropdownMenu(
+    searchFilterItemUiState: SearchFilterItemUiState<T>,
+    onSelected: (T, Boolean) -> Unit,
+    filterChipLabelDefaultText: String,
+    dropdownMenuItemText: (T) -> String,
```

Comment:
Currently, the only parameter that differs in FilterChip is the text to be displayed.

So I prepared a composable that passes the text and use it.